### PR TITLE
fix(rpc): validators endpoint fail during quorum rotation

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

During quorum rotation, RPC requests to `/validators` fail with error:

```
{
  "code": -32603,
  "message": "Internal error",
  "data": "failed to create validator scoring strategy: could not initialize proposer: quorum hash mismatch at height 33226"
}
```

## What was done?

1. Fixed the issue 
2. Added logger to state store

## How Has This Been Tested?

Modified e2e tests to chain halt on rotation and executed the /validators query.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced logging capabilities for the state store, improving traceability during operations.
	- Added a new method to create a state store instance with a custom logger.
	- Implemented a new test case to validate loading validators during quorum rotations.

- **Bug Fixes**
	- Improved error logging for proposer selection, providing better context during quorum rotations.
	- Enhanced error handling and messaging for loading validators and consensus parameters.

- **Documentation**
	- Expanded comments for clarity in key methods related to proposer selection and state store initialization.

### CODE REVIEW

File: internal/state/store.go [hunk 1]
```diff
@@ -562,27 +581,48 @@
...
+		if err != nil {
+			return nil, fmt.Errorf("failed to create validator scoring strategy: %w", err)
+		}
+
+		// now, advance to (height,0)
+		if err := strategy.UpdateHeightRound(height, 0); err != nil {
+			return nil, fmt.Errorf("failed to update validator scores at height %d, round 0: %w", height, err)
+		}
+
+		return strategy.ValidatorSet(), nil
```
[the remark this answers is not in the document]
Comment (lklimek) [reply internal/state/store.go]: removed quorum rotation log

File: internal/consensus/versioned/selectproposer/height_proposer.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func (s *heightProposerSelector) proposerFromStore(height int64) error {
 				"validators_hash", meta.Header.ValidatorsHash, "quorum_hash", s.valSet.QuorumHash,
 				"validators", s.valSet)
 
-			return fmt.Errorf("quorum hash mismatch at height %d", height)
+			return fmt.Errorf("validator set hash mismatch at height %d", height)
```
[the remark this answers is not in the document]
Comment (lklimek) [reply internal/consensus/versioned/selectproposer/height_proposer.go:93]: The fix eliminates this error, so it should not happen anymore.

File: internal/state/store.go [hunk 3]
```diff
@@ -122,8 +122,12 @@ type dbStore struct {
 var _ Store = (*dbStore)(nil)
 
 // NewStore creates the dbStore of the state pkg.
-func NewStore(db dbm.DB) Store {
-	return dbStore{db, log.NewNopLogger()}
+func NewStore(db dbm.DB, logger ...log.Logger) Store {
```
Comment (shumkov) [inline internal/state/store.go:125]: Please explain with a comment in the code what is happening here

**Review by shumkov — CHANGES_REQUESTED** [review]

File: internal/state/store.go (hunk 3 above)
Comment (lklimek) [reply internal/state/store.go:135]: done

**Review by shumkov — APPROVED** [review]

**Pull request merged**